client/my-sites/controller.js

```diff
@@ -517,6 +518,14 @@ export function noSite( context, next ) {
 	return next();
 }
 
+// Prefetch the site data that is not from the `/sites` or `/sites/:site` endpoint.
+function prefetchSiteData( context, next ) {
+	const state = context.store.getState();
+	const selectedSite = getSelectedSite( state );
+
+	context.store.dispatch( requestAdminColor( selectedSite.ID ) ).then( next ).catch( next );
```

Comment:
Okay, on my testing, the issue only happens when we just recently changed the admin color scheme of a site (from `/wp-admin/profile.php`), and then open a Calypso nav-unified page (e.g. `/plans`) of that site for the first time. The previously cached color scheme will be flashed and then the new color scheme will be applied (then cached again). Then, this issue won't happen anymore (until you change the color scheme again).

With this change, however, we will always synchronously block the rendering of all pages on every load with an API request to get the admin color. I'm not convinced that this slowdown (even just a bit) is justified by this issue. Let's not further slow down any existing slowness in Calypso.

What if we just do nothing? I think the user will understand that the flashed color is the previous color scheme. Also, changing color scheme should happen very very rarely.

cc: @davemart-in

Reply:
I agree. It's not worth it, but at least we now have this documented. Going to cc @ianstewart to see if he has a different take though.